synapse/lib/structlog.py

```diff
@@ -0,0 +1,41 @@
+import json
+
+import logging
+
+import synapse.common as s_common
+
+class JsonFormatter(logging.Formatter):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    def format(self, record: logging.LogRecord):
+
+        record.message = record.getMessage()
+        mesg = self.formatMessage(record)
+        ret = {
+            'message': mesg,
+            'logger': {
+                'name': record.name,
+                'thread': record.threadName,
```

Comment:
are thread/process names going to be worth the extra volume?

Reply:
thread name removed